### PR TITLE
fix(design-tokens): improve consistency across the design system

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #9524

## Summary

- [x] Deprecate the following from semantic > font:
  - [x]  letter-spacing
  - [x]  paragraph-spacing
  - [x]  text-decoration
  - [x]  text-case
- [x] Reassign font > font-weight > bold in semantic tokens to {core.font.weight.demi}
- [x] Make sure date-picker-month .week-header uses --calcite-font-weight-bold

### CODE REVIEW

**Comment by jcfranco** [comment]
@alisonailea There are some [test failures](https://github.com/Esri/calcite-design-system/actions/runs/11559059110/job/32172878737?pr=10635#step:5:191) related to `calcite-design-tokens` imports. 

```
@esri/calcite-components:build: [ ERROR ]  TypeScript: src/utils/responsive.ts:7:8
@esri/calcite-components:build:            Cannot find module '@esri/calcite-design-tokens/dist/es6/global' or
@esri/calcite-components:build:            its corresponding type declarations.
@esri/calcite-components:build: 
@esri/calcite-components:build:       L6:    calciteContainerSizeWidthXxs,
@esri/calcite-components:build:       L7:  } from "@esri/calcite-design-tokens/dist/es6/global";
@esri/calcite-components:build: 
@esri/calcite-components:build: [ ERROR ]  TypeScript: src/components/date-picker-month-header/date-picker-month-header.tsx:17:8
@esri/calcite-components:build:            Cannot find module '@esri/calcite-design-tokens/dist/es6/global.js'
@esri/calcite-components:build:            or its corresponding type declarations.
@esri/calcite-components:build: 
@esri/calcite-components:build:      L16:    calciteSizeSm,
@esri/calcite-components:build:      L17:  } from "@esri/calcite-design-tokens/dist/es6/global.js";
@esri/calcite-components:build:      L18:  import {
@esri/calcite-components:build: 
@esri/calcite-components:build: [16:41.8]  build failed in 17.54 s
```

Can you take a closer look?

File: packages/calcite-design-tokens/src/semantic/font.json [hunk 1]
```diff
@@ -663,254 +663,6 @@
             }
           }
         }
-      },
-      "letter-spacing": {
```
Comment (jcfranco) [inline packages/calcite-design-tokens/src/semantic/font.json:667]: Would this be breaking due to these removals?

Comment (alisonailea) [reply packages/calcite-design-tokens/src/semantic/font.json:667]: technically yes but they were not used by any components.

Comment (jcfranco) [reply packages/calcite-design-tokens/src/semantic/font.json:667]: They are mentioned in the design token [reference doc](https://developers.arcgis.com/calcite-design-system/foundations/tokens/reference/).

**Comment by jcfranco** [comment]
@alisonailea Per [our conventions](https://github.com/Esri/calcite-design-system/blob/main/CONTRIBUTING.md#breaking-changes), can you add a `BREAKING CHANGE` footer to the PR description with more details behind these changes? 

@geospatialem Is it possible to include this in 3.0? #9524 wasn't labeled as breaking and therefore isn't included in our planned breaking changes doc. Some of these removals are covered in the [design token reference doc](https://developers.arcgis.com/calcite-design-system/foundations/tokens/reference/#shadow), so it could impact some teams without a deprecation period.

**Comment by geospatialem** [comment]
> @geospatialem This is probably too late to include in 3.0, right? #9524 wasn't labeled as breaking and therefore isn't included in our planned breaking changes doc. Some of these removals are covered in the [design token reference doc](https://developers.arcgis.com/calcite-design-system/foundations/tokens/reference/#shadow), so it could impact some teams without a deprecation period.

@jcfranco Agree that we cannot deliver this as a breaking change for `3.0` as it has not been communicated to teams.

Was under the same impression Franco mentions that this would be a visual change only, and not breaking. @alisonailea is there a way we could deprecate the tokens for `3.0`, and add this to our `4.0` doc for removal?

**Comment by alisonailea** [comment]
@jcfranco and @geospatialem totally I'm sorry I forgot that. I've updated the PR to only deprecate the font tokens and removed the breaking change.

File: packages/calcite-design-tokens/src/semantic/font.json (hunk 1 above)
Comment (alisonailea) [reply packages/calcite-design-tokens/src/semantic/font.json]: tokens are updated to be deprecated only

**Comment by jcfranco** [comment]
@alisonailea Thanks for updating!

@geospatialem Do the deprecations need to be separated and use the new `deprecate` commit type?

**Comment by alisonailea** [comment]
Closing in favor of separate PRs #10742 #10743

**Pull request closed without merging**